b9lab-content/2-what-is-cosmos/03-cosmos-ecosystem.md

```diff
@@ -0,0 +1,177 @@
+
+A second idea, one that **reduces the number of connections necessary** back down to the number of chains, would be to introduce an intermediate entity that facilitates the transfer between chains, in a hub and spoke model. Unfortunately, this would bring back unwelcome centralization.
+
+*How could we uphold decentralization then?*
+
+First, by keeping the capability of chain-to-chain connections made possible by **IBC**. Second, by creating a flexible multi-hub and spoke model whereby a spoke gets automatic access to all the other spokes of the hub it connected to. This is the road **Cosmos** took.
+
+Cosmos implements a **modular architecture with two blockchain classes**: **hubs** and **zones**.
+
+**Zones** are heterogenous blockchains carrying out the authentication of accounts and transactions, the creation and distribution of tokens, and the execution of changes to the chain.
+
+**Hubs** connect these so-called zones. Hubs are blockchains designed to connect heterogenous blockchains, i.e. zones. Once a zone connects to a hub through an IBC connection, it gets automatic access to the other zones connected to the hub. At this point, data and value can be sent and received between the zones without risk, for instance of double-spending tokens. This helps reduce the number of chain-to-chain connections that need to be established for interoperability.
+
+--> Include hotspot image on zones and hubs
+
+An example of such a specific chain is the **[Cosmos Hub](https://hub.cosmos.network/main/hub-overview/overview.html)**, the first hub to be created. It is a public, Proof-of-Stake (PoS) blockchain with a native token, ATOM. The Cosmos Hub can be understood as a router facilitating, among other things, transactions between the chains connected to it. For example, transaction fees can be paid in different tokens as long as the zone trusts the Cosmos Hub and the other zones connected to it.
+
+*How do we connect our chain to a non-Tendermint chain?* In fact, the IBC connection is not limited to Tendermint-based chains.
+
+If the other non-Tendermint chain is a **fast-finality chain**, i.e. a blockchain using a fast-finality consensus algorithm, the connection can be established by adapting IBC to work with the non-Tendermint consensus mechanism.
+
+If the other chain is a **probabilistic-finality chain**, i.e. a blockchain without fast finality like those relying on Proof-of-Work (PoW), a simple adaptation of IBC is not sufficient. Instead, a more elaborate setup is required. More precisely, a proxy-chain, a so-called **peg-zone**, helps establish interoperability. Peg-zones are fast-finality blockchains tracking chain states to establish finality. The peg-zone chain itself is IBC-compatible and acts as the **bridge** between the rest of the IBC network, i.e. its blockchains, and the probabilistic-finality chain.
```

Comment:
Not quite true. The real limit is the ability of of the other chain to implement the core IBC standards (primarily ICS02, 03, 04).

Reply:
instead of these two paragraphs about fast finality or not I would just say the only requirement is that it be possible to implement a *light client* for the chain. Then we only need to implement the ICS02 client interface for that chains light client and it can connect over IBC. of course additionally the chain will need the ability to implement the rest of IBC logic, so it might need some smart contracting facility. so bitcoin for instance, even though it has a light client, it doesnt have enough functionality to allow you to implemenet IBC on Bitcoin. But with Ethereum, you can implement IBC in solidity and do a full Cosmos<->Ethereum IBC connection. But with Avalanche, even though it's fast finality, we don't know how to do a light client, so we're not sure how to do IBC yet ... hope that helps!

Reply:
Thank you @ebuchman! That definitely helps.

Reply:
Is this proposal acceptable? 

1. It must be possible to implement a *light client* that is capable of exploring and interacting with the chain without necessarily syncing the entire chain. 
2. It must be possible to implement the [ICS02 client interface](https://github.com/cosmos/ibc/blob/master/spec/core/ics-002-client-semantics/README.md) with the light client. The ICS02 client interface usually requires smart contract capability on the other chain where important parts of the interface will be implemented. For this reason, most, but not all, blockchains with light client capability are theoretically compatible with IBC. For example, Ethereum has both light client and smart contract capability, so and IBC bridge to Ethereum is feasible.